strawberryfields/backends/tfbackend/ops.py

```diff
@@ -1464,7 +1455,9 @@ def conditional_state(system, projector, mode, state_is_pure, batched=False):
     einsum_args = [system, tf.math.conj(projector)]
     if not state_is_pure:
         einsum_args.append(projector)
-    cond_state = tf.einsum(eqn, *einsum_args)
+
+    # does not work with `tf.einsum`; are the `einsum_args` shapes wrong?
+    cond_state = _einsum_v1(eqn, *einsum_args)
```

Comment:
This _should_ work using `tf.einsum`, but there were issues with the shape of the `einsum_args` arrays. It looked like they were of the wrong shape, but I couldn't figure it out. It works (for now) using `_einsum_v1` as before, but it'd be nicer to simply use `tf.einsum` instead.

Reply:
Unless the shape of the input tensors in changed, I there's no way around the issue. You're correct in the sense that the shape is the problem, from the new `einsum` (`_einsum_v2`) [code comments](https://github.com/tensorflow/tensorflow/blob/d8ce9f9c301d021a69953134185ab728c1c248d3/tensorflow/python/ops/special_math_ops.py#L1187-L1191)

```
    # Validate and sanitize the equation and resolve static input shapes, as
    # opt_einsum requires that all shapes be a tuple of positive integers.
    # Also remove ellipsis from the equation as opt_einsum will replace them
    # with named labels. Then broadcasting between different shapes or ranks
    # wouldn't work. (E.g. [1, 1, 2] wouldn't broadcast with [3, 1]).
```

Some of the inputs are, for example, of sizes [1, 6, 6, 6, 6, 6, 6] and [1, 6], and then won't broadcast and work with `einsum`.